packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -383,28 +388,82 @@ class RenderAspectRatio extends RenderProxyBox {
     markNeedsLayout();
   }
 
+  double getMinIntrinsicWidth(BoxConstraints constraints) {
+    return constraints.minWidth;
+  }
+
+  double getMaxIntrinsicWidth(BoxConstraints constraints) {
+    return constraints.maxWidth;
+  }
+
   double getMinIntrinsicHeight(BoxConstraints constraints) {
-    return _applyAspectRatio(constraints).height;
+    return constraints.minHeight;
   }
 
   double getMaxIntrinsicHeight(BoxConstraints constraints) {
-    return _applyAspectRatio(constraints).height;
+    return constraints.maxHeight;
   }
 
   Size _applyAspectRatio(BoxConstraints constraints) {
     assert(constraints.debugAssertIsNormalized);
-    double width = constraints.constrainWidth();
-    double height = constraints.constrainHeight(width / _aspectRatio);
-    return new Size(width, height);
-  }
+    assert(() {
+      if (!constraints.hasBoundedWidth && !constraints.hasBoundedHeight) {
+        throw new RenderingError(
+          'This $runtimeType was given an aspect ratio of $aspectRatio but was given '
```

Comment:
Make the first line be a relatively short "error message" followed by a `\n`. e.g.:
```
'$runtimeType has unbounded constraints.\n'
```